metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -269,15 +288,24 @@ public void close() throws Exception {
 
     // Events handled by Migration Driver.
     abstract class MigrationEvent implements EventQueue.Event {
+        @SuppressWarnings("ThrowableNotThrown")
         @Override
         public void handleException(Throwable e) {
-            if (e instanceof RejectedExecutionException) {
-                log.info("Not processing {} because the event queue is closed.", this);
+            if (e instanceof MigrationClientAuthException) {
+                KRaftMigrationDriver.this.faultHandler.handleFault("Encountered client auth error in " + this, e);
```

Comment:
maybe "Encountered ZooKeeper authentication error" to be as specific as possible?